### PR TITLE
Add helper functions to handle kubeconfig operations

### PR DESCRIPTION
**Pull request by marioreggiori**
### Description

Add helper functions to handle kubeconfig operations.

### Checklist

* [x] added release notes to `Unreleased` section in [CHANGELOG.md](CHANGELOG.md), if user facing change

### References
Based on #151 

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: pkg/apis/kubernetes/v1/kubeconfig_genclient.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	req.Body = nil
+	req.ContentLength = 0
+
+	return req, nil
+}
+
+func (k *kubeconfig) FilterAPIResponse(ctx context.Context, res *http.Response) (*http.Response, error) {
+	// TODO: do we care if this fails? ... not sure how to cover the error branch without mocking a io.ReadCloser
```
Comment (LittleFox94) [inline pkg/apis/kubernetes/v1/kubeconfig_genclient.go:59]: hm I think you can ignore that error

File: pkg/apis/kubernetes/v1/kubeconfig_helper.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	cluster := Cluster{Identifier: clusterID}
+
+	for {
+		if err := a.Get(ctx, &cluster); err != nil {
+			return "", fmt.Errorf("failed to get cluster: %w", err)
+		}
+
+		if cluster.KubeConfig != "" {
+			// TODO: maybe check the validity and request new if already expired?
```
Comment (LittleFox94) [inline pkg/apis/kubernetes/v1/kubeconfig_helper.go:28]: not expiring and not go-anxclouds task to check

File: pkg/apis/kubernetes/v1/kubeconfig_types.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package v1
+
+type kubeconfig struct {
+	Cluster string `json:"cluster" anxcloud:"identifier"`
+}
```
Comment (LittleFox94) [inline pkg/apis/kubernetes/v1/kubeconfig_types.go:5]: I think this could go into `kubeconfig_helper.go` as well, being an internal type and all

**Review by LittleFox94 — APPROVED** [review]

File: pkg/apis/kubernetes/v1/crud_test.go [hunk 4]
```diff
@@ -295,6 +297,80 @@ var _ = Describe("CRUD", Ordered, func() {
 			Expect(err).To(MatchError(api.ErrNotFound))
 		})
 	})
+
+	Context("KubeConfig", func() {
+		BeforeEach(func() {
+			if isIntegrationTest {
```
[the remark this answers is not in the document]
Comment (marioreggiori) [reply pkg/apis/kubernetes/v1/crud_test.go]: Moved pure mock tests to dedicated `kubeconfig_test.go`

**Review by LittleFox94 — APPROVED** [review]

**Pull request merged**